mozldap.go

```diff
@@ -83,49 +74,44 @@ func NewClient(uri, username, password string, tlsconf *tls.Config, starttls boo
 		cli.conn, err = ldap.Dial("tcp", fmt.Sprintf("%s:%d", cli.Host, cli.Port))
 	}
 	if err != nil {
-		panic(err)
+		return Client{}, fmt.Errorf("%s -> %s", errorText, err.Error())
 	}
 	// TLS and StartTLS are mutually exclusive
 	if !cli.UseTLS && starttls {
 		cli.UseStartTLS = true
 		err = cli.conn.StartTLS(tlsconf)
 		if err != nil {
-			cli.conn.Close()
-			panic(err)
+			cli.Close()
```

Comment:
I could change it back, but I don't see a particular reason to. The method exists in the API, might as well use it in the library.